forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/registrationrequest/DirectorySoftwareStatementFactory.java

```diff
@@ -0,0 +1,162 @@
+                    getCompetentAuthorityClaims(softwareStatementClaims));
+        }
+
+        softwareStatement = softwareStatementBuilder.build();
+
+        return softwareStatement;
+    }
+
+    private OrganisationAuthorityClaims getCompetentAuthorityClaims(
+            RegistrationRequestJWTClaims softwareStatementClaims) throws ParseException {
+        JSONObject contactsJsonArray =
+                (JSONObject) softwareStatementClaims.getClaim( SSAClaims.ORGANISATION_COMPETENT_AUTHORITY_CLAIMS);
+        return new OrganisationAuthorityClaims.OrganisationAuthorityClaimsBuilder()
+                .authority_id(JSONObjectUtils.getString(contactsJsonArray, SSAClaims.OCAC_AUTHORITY_ID))
+                .registration_id(JSONObjectUtils.getString(contactsJsonArray, SSAClaims.OCAC_REGISTRATION_ID))
+                .status(JSONObjectUtils.getString(contactsJsonArray, SSAClaims.OCAC_STATUS))
+                .authorizations(getAuthorizations(contactsJsonArray)).build();
+    }
+
+    private List<AuthorisationClaim> getAuthorizations(JSONObject contactsJsonArray) throws ParseException {
+        List<AuthorisationClaim> authClaims = new ArrayList<>();
+        JSONArray authorityClaims = (JSONArray) contactsJsonArray.get(SSAClaims.OCAC_AUTHORISATIONS);
+        for (Object authorityClaim : authorityClaims) {
+            JSONObject authorityClaimJson = (JSONObject) authorityClaim;
+            AuthorisationClaim authClaim = new AuthorisationClaim.AuthorisationClaimBuilder()
+                    .member_state(JSONObjectUtils.getString(authorityClaimJson, SSAClaims.OCAC_MEMBER_STATE))
+                    .roles(JSONObjectUtils.getStringList(authorityClaimJson, SSAClaims.OCAC_ROLES)).build();
+            authClaims.add(authClaim);
+        }
+        return authClaims;
+    }
+
+    private List<OrganisationContact> getOrgContacts(RegistrationRequestJWTClaims softwareStatementClaims) {
+        List<OrganisationContact> contacts = new ArrayList<>();
+        JSONArray contactsJsonArray = (JSONArray) softwareStatementClaims.getClaim(SSAClaims.ORG_CONTACTS);
+        if (contactsJsonArray != null) {
+            for (Object contactJson : contactsJsonArray) {
+                JSONObject contactJsonObject = ((JSONObject) contactJson);
+                OrganisationContact orgContact =
+                        new OrganisationContact.OrganisationContactBuilder()
+                                .name(getContactField(contactJsonObject, "name"))
+                                .email(getContactField(contactJsonObject, "email"))
+                                .type(getContactField(contactJsonObject, "type"))
+                                .phone(getContactField(contactJsonObject, "phone")).build();
+                contacts.add(orgContact);
+            }
+        } else {
+            String errorString = "No " + SSAClaims.ORG_CONTACTS + " field available from Software Statement";
+            log.info("getOrgContacts() {}", errorString);
```

Comment:
Could be better log.error or log.warn instead of log.info? because contains the error message.